### PR TITLE
feat: support deletion for memory resolver

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Part 2/4 of #454 

Based on draft PR #582

### CODE REVIEW

File: internal/resolver/memory.go [hunk 1]
```diff
@@ -48,6 +48,11 @@ func (m *Memory) Tag(_ context.Context, desc ocispec.Descriptor, reference strin
 	return nil
 }
 
+// Delete removes a reference from index map.
+func (m *Memory) Delete(reference string) {
```
Comment (Wwwsylvia) [inline internal/resolver/memory.go:52]: Just a thoughts: should this be called `Delete` or `UnTag`? 🤔

Comment (shizhMSFT) [reply internal/resolver/memory.go:52]: +1 on `Untag`.

Comment (wangxiaoxuan273) [reply internal/resolver/memory.go:52]: renamed it as `Untag`.

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Pull request merged**